rpcs3/rpcs3qt/instruction_editor_dialog.cpp

```diff
@@ -13,17 +13,33 @@ constexpr auto qstr = QString::fromStdString;
 
 extern bool ppu_patch(u32 addr, u32 value);
 
-instruction_editor_dialog::instruction_editor_dialog(QWidget *parent, u32 _pc, const std::shared_ptr<cpu_thread>& _cpu, CPUDisAsm* _disasm)
+instruction_editor_dialog::instruction_editor_dialog(QWidget *parent, u32 _pc, CPUDisAsm* _disasm, std::function<cpu_thread*()> func, instruction_editor_dialog** storage_pointer)
 	: QDialog(parent)
 	, m_pc(_pc)
 	, m_disasm(_disasm)
-	, m_cpu(_cpu)
+	, m_get_cpu(std::move(func))
 {
 	setWindowTitle(tr("Edit instruction"));
 	setAttribute(Qt::WA_DeleteOnClose);
 	setMinimumSize(300, sizeHint().height());
 
-	m_cpu_offset = m_cpu->id_type() == 2 ? static_cast<spu_thread&>(*m_cpu).ls : vm::g_sudo_addr;
+	if (storage_pointer)
+	{
+		connect(this, &QDialog::finished, [storage_pointer](int)
+		{
+			*storage_pointer = nullptr;
+		});
+	}
+
+	const auto cpu = m_get_cpu();
+
+	if (!cpu)
+	{
+		close();
```

Comment:
this is horrible...
Never close a dialog in the constructor.

Reply:
Qt events are not processed here, the constructor will not call tthe destructor himself. So why not?

Reply:
Because the object itself gets destroyed and can't be used -> crash

Reply:
Made show() not be called after this case, so it's safe.